File: ExercismFSharp/ExercismFSharp/GradeSchool.fs
module GradeSchool

type School = { roster : (int * string) list }

let empty = { roster = [] }

let add name grade school =
    { roster = (grade, name) :: school.roster }

let grade grade school =
    school.roster
    |> List.filter (fun (g, _) -> g = grade)
    |> List.map snd
    |> List.sort

let roster school =
    school.roster
    |> List.groupBy fst
    |> List.map (fun (grade, values) ->
        let namesInGrade =
            values
            |> List.map snd
            |> List.sort
        (grade, namesInGrade)
    )
    |> List.sort

File: Exercism/python/grade-school/grade_school.py
import collections


class School(object):
    def __init__(self):
        self.enrolled = collections.defaultdict(list)

    def add_student(self, name, grade):
        self.enrolled[grade].append(name)
        self.enrolled[grade].sort()

    def roster(self):
        return [name
                for grade, names in sorted(self.enrolled.items())
                for name in sorted(names)
            ]

    def grade(self, grade_number):
        return self.enrolled.get(grade_number, [])
